app.js

```diff
@@ -10,84 +11,181 @@ app.use(express.static('public'));
 
 const server = app.listen(port, () => {
-  console.log(`Getlabs sample application listening at http://localhost:${port}`)
+    console.log(`Getlabs sample application listening at http://localhost:${port}`)
 });
 
+// config provided by getlabs
 const getlabsConifg = {
-  hostname: 'sandbox.api.getlabs.com',
-  apiToken: ''
-}
-
-if (!getlabsConifg.apiToken) {
-  throw new Error('Missing API Token. Add your Getlabs API token to getlabsConifg.');
+    hostname: 'sandbox.api.getlabs.com',
+    apiToken: '', // this is your api token provided by getlabs
+    jwt: {
+        sub: '', // this is your getlabs provided client id
+        aud: 'https://api-external.getlabs.com/oauth/token',
+    }
 }
 
 // Simulates a logged in user. All API requests will be made for this patient.
 const loggedInUser = {
-  email: '', // email address will receive notifications
-  dob: '1970-01-01',
-  birthSex: 'female',
-  phoneNumber: '', // ex: 4805551234, phone number will receive SMS notifications
-  firstName: 'Test',
-  lastName: 'Patient'
+    email: '', // email address will receive notifications
+    dob: '1970-01-01',
+    birthSex: 'female',
+    phoneNumber: '', // ex: 4805551234, phone number will receive SMS notifications
+    firstName: 'Test',
+    lastName: 'Patient'
 }
 
+// check getlabsConifg is setup
 if (!getlabsConifg.apiToken) {
-  throw new Error('Missing API Token. Add your Getlabs API token to getlabsConifg.');
+    throw new Error('Missing API Token. Add your Getlabs API token to getlabsConifg.');
+} else if (!getlabsConifg.jwt.sub) {
+    throw new Error('Missing sub from jwt. Add your Getlabs client id to getlabsConifg.jwt.sub.');
 } else if (!loggedInUser.email || !loggedInUser.phoneNumber) {
-  throw new Error('Missing user email address or phone number. Add the missing data to loggedInUser.');
+    throw new Error('Missing user email address or phone number. Add the missing data to loggedInUser.');
+}
+
+/**
+ * this houses the oauth token response from getlabs api
+ * @type {{}}
+ */
+let oauthTokenResp = {
+    access_token: '',
+    refresh_token: ''
 }
 
-const agent = superagent
-  .agent()
-  .use(superagentPrefix(`https://${getlabsConifg.hostname}`))
-  .set('accept', 'json')
-  .set('Authorization', `Bearer ${getlabsConifg.apiToken}`);
+/**
+ * returns the http client used to make external requests
+ * @param bearerToken: the Authorization bearer token value for said request
+ */
+const getHttpClient = (bearerToken) => superagent
+    .agent()
+    .use(superagentPrefix(`https://${getlabsConifg.hostname}`))
+    .set('accept', 'json')
+    .set('Authorization', `Bearer ${bearerToken}`);
+
+/**
+ * this generates a signed jwt in order for us to receive an oauth access_token from getlabs
+ * NOTE: we sign the jwt with the apiToken as your secret
+ */
+const generateJwtForOauthToken = () => {
+    const jwt = require('jsonwebtoken');
+
+    // build header
+    const jwtHeader = {
+        algorithm: 'HS256',
```

Comment:
I'm wondering if we should consider using HS512 here instead. On top of the added security of a longer key, I believe SHA512 is actually faster on 64 bit CPUs than SHA256. The overall length of the encoded token should only have a small increase.